src/ui/components/form/InputBn.tsx

```diff
@@ -8,29 +8,58 @@ import { ArgComponentProps } from 'types';
 
 type Props = ArgComponentProps<BN>;
 
-export function InputBn({ onChange, typeDef }: Props): JSX.Element {
+function getMinMax(type: string): [bigint, bigint] {
+  switch (type) {
+    case 'i8':
+      return [-128n, 127n];
+    case 'i16':
+      return [-32768n, 32767n];
+    case 'i32':
+      return [-2147483648n, 2147483647n];
+    case 'i64':
+      return [-9223372036854775808n, 9223372036854775807n];
+    case 'i128':
+      return [-170141183460469231731687303715884105728n, 170141183460469231731687303715884105727n];
+    case 'u8':
+      return [0n, 255n];
+    case 'u16':
+      return [0n, 65535n];
+    case 'u32':
+      return [0n, 4294967295n];
+    case 'u64':
+      return [0n, 18446744073709551615n];
+    case 'u128':
+      return [0n, 340282366920938463463374607431768211455n];
+    default:
+      return [0n, BigInt(Number.MAX_SAFE_INTEGER)];
+  }
+}
+
+export function InputBn({ onChange, typeDef: { type } }: Props): JSX.Element {
   const [displayValue, setDisplayValue] = useState('0');
+  const [min, max] = getMinMax(type);
 
   const handleChage = useCallback(
-    (e: React.ChangeEvent<HTMLInputElement>) => {
-      if (e.target.value.trim()) {
-        const bn = new BN(e.target.value);
-        setDisplayValue(e.target.value);
-        onChange(bn);
+    ({ target: { value } }: React.ChangeEvent<HTMLInputElement>) => {
+      if (value.trim()) {
+        const val = Number(value);
+        if (min <= val && val <= max) {
```

Comment:
Wouldn't it be cool to do this in the "validation" part?

This way no error message is shown. Couldn't we probably also used `zod` here to add some validation functions to our forms with proper error messages. 🤔

Reply:
after using zod in the Multisig, i wouldn't use it again or recommend it to anyone. we definitely need a better way to handle validation, but this is not the scope of this PR. (a large refactoring of all things form related is needed)
I could add some validation messages but I was thinking it's ok as is because our users are developers and these min and max values are expected.